linkerd/proxy/detect/src/lib.rs

```diff
@@ -83,85 +40,40 @@ impl<D: Clone, A> tower::layer::Layer<A> for DetectProtocolLayer<D> {
     fn layer(&self, accept: A) -> Self::Service {
         Self::Service {
             detect: self.detect.clone(),
-            peek_capacity: self.peek_capacity,
             accept,
         }
     }
 }
 
 impl<T, D, A> tower::Service<(T, BoxedIo)> for DetectProtocol<D, A>
 where
-    D: Detect<T>,
-    A: core::listen::Accept<(D::Target, BoxedIo)> + Clone,
-    D::Target: std::fmt::Debug,
+    T: Send + 'static,
+    D: Detect<T> + Clone + Send + 'static,
+    D::Target: Send,
+    A: core::listen::Accept<(D::Target, BoxedIo)> + Send + Clone + 'static,
+    A::Future: Send,
 {
     type Response = A::ConnectionFuture;
     type Error = Error;
-    type Future = AcceptFuture<T, D, A>;
+    type Future = Pin<Box<dyn Future<Output = Result<A::ConnectionFuture, Error>> + Send>>;
 
-    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        self.accept.poll_ready(cx).map_err(Into::into)
+    fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        // The `accept` is cloned into the response future, so its readiness isn't important.
+        Poll::Ready(Ok(()))
```

Comment:
this does seem like a slight change in behavior, but i don't think it's one that actually matters. is there a reason we changed this? it seems like we could continue driving it here.

Reply:
I think it was technically a bug before. We would poll the accept to ready, and then clone it, and then not poll the clone to ready (which is technically required).

I think, practically, this stack doesn't use readiness and there's a oneshot inside of it.